Add Collaboration Diagram-Customer

SanyaDua opened the pull request.

SanyaDua committed:
Merge pull request #2 from zacaz90/master

Merge from  master

SanyaDua committed:
Merge pull request #3 from SanyaDua/master

Merge to develop

SanyaDua committed:
Add Customer Collaboration Diagram

seant1 reviewed (commented):
Looks good!

zacaz90 commented:
I didn't get a chance to review this before it was merged, my fault.
There were a couple of changes to the class diagram you mentioned, some of which were instant adds - thanks for the feedback!
Some of them I just want to bounce back with:
* "RedirectInboundCustomer()" I've got instead as redirectCustomer(String destination). I think since Customer was considered one class and not separated into two we could make a more general method for it.
* "inputDetails()" I've got inputOption() but I see what you mean. If they leave their phone number to call back or something they need to inputDetails() as well as the option in response to the voice unit. Maybe worth putting both in? Definitely added to the class diagram though.
* "voiceUnitgivesCMCDetails()" could probably be a more general giveCustomerDetails() as it's shown to be coming from the voice unit to the CMC. Somehow I totally missed it anyway... consider it added to the class diagram!
* "paymentConfirmationText()" I think should be givePaymentConfirmationText() or display or show, I think it needs to begin with a verb
* "tellTravelPrefrences()" wow good catch that's mighty hard to find, should be Preferences
* As for "askPaymentDetails()" and my "requestPaymentDetails()" I'm not hugely fussed, I'm just trying to suck up to the markers ahahaha - let me know what you think
* Also I don't quite understand "pointedToRM()" which I believe we changed from "route". Happy to change it I just want to understand it a bit better if possible.
They're all good points and anything I didn't mention was an indisputable addition, so thanks again. You'll have to add the parameters into any functions that need it too. I'll wait to see what you say before I update the rest of the class diagram so I don't have to PR twice or anything

SanyaDua commented:
Thanks Zac for that feedback, I have updated the following changes in the customer collaboration diagram:

redirectCustomer instead of inbound

giveCustomerDetails()

givePaymentConfirmationText()

tellTravelPrefrences()- I just fixed the spelling error 
 
requestPaymentDetails

routeToRM()" - I changed it back to this

Thanks, let me know if you have any other things in mind to change.
